vpr/src/place/place_constraints.cpp

```diff
@@ -257,3 +258,95 @@ void load_cluster_constraints() {
         }
     }
 }
+
+void mark_fixed_blocks() {
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& place_ctx = g_vpr_ctx.mutable_placement();
+    auto& floorplanning_ctx = g_vpr_ctx.floorplanning();
+
+    for (auto blk_id : cluster_ctx.clb_nlist.blocks()) {
+        if (!is_cluster_constrained(blk_id)) {
+            continue;
+        }
+        PartitionRegion pr = floorplanning_ctx.cluster_constraints[blk_id];
+        auto block_type = cluster_ctx.clb_nlist.block_type(blk_id);
+        t_pl_loc loc;
+
+        if (is_pr_size_one(pr, block_type, loc)) {
+            //Set block location and grid usage
+            set_block_location(blk_id, loc);
+
+            //Set as fixed
+            place_ctx.block_locs[blk_id].is_fixed = true;
+        }
+    }
+}
+
+bool is_region_size_one(const Region& reg, t_logical_block_type_ptr block_type) {
+    auto& device_ctx = g_vpr_ctx.device();
+    vtr::Rect<int> rb = reg.get_region_rect();
+    bool size_one = false;
+
+    if (rb.xmin() == rb.xmax() && rb.ymin() == rb.ymax()) {
+        if (reg.get_sub_tile() != NO_SUBTILE) {
```

Comment:
Don't need to put an error message, but comment saying that later on the code will catch is subtile is invalid at this location

Reply:
check whether the the subtile is compatible at this tile location -> if it is, return true, if not, return false

Reply:
essentially, this routine returns false if it is bigger than one or less than one

Reply:
to differentiate the cases, you could make an enumerated type - reg covers 0, 1, or more than one loc. return this type instead of a bool